From d20c559dbe78236d298da77f22ae7316f347dbd3 Mon Sep 17 00:00:00 2001
From: Mathieu Schroeter <mathieu@schroetersa.ch>
Date: Sun, 1 May 2016 11:59:43 +0200
Subject: [PATCH 2/4] Don't forward logs to files

---
 cmake/FindWpkgBuild.cmake | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/cmake/FindWpkgBuild.cmake b/cmake/FindWpkgBuild.cmake
index 45cd5b1..574c758 100644
--- a/cmake/FindWpkgBuild.cmake
+++ b/cmake/FindWpkgBuild.cmake
@@ -182,8 +182,8 @@ function( unigw_ConfigureMakeProject )
 	add_custom_target(
 		${FULL_PROJECT_NAME}-make
 		COMMAND ${ECHO_CMD} ${BUILD_CMD}
-			1> ${BUILD_DIR}/${FULL_PROJECT_NAME}_make.log
-			2> ${BUILD_DIR}/${FULL_PROJECT_NAME}_make.err
+			# 1> ${BUILD_DIR}/${FULL_PROJECT_NAME}_make.log
+			# 2> ${BUILD_DIR}/${FULL_PROJECT_NAME}_make.err
 		DEPENDS ${CONFIGURE_TARGETS}
 		WORKING_DIRECTORY ${BUILD_DIR}
 		COMMENT "Building ${FULL_PROJECT_NAME}"
@@ -193,8 +193,8 @@ function( unigw_ConfigureMakeProject )
 	add_custom_target(
 		${FULL_PROJECT_NAME}-package
 		COMMAND ${ECHO_CMD} ${BUILD_CMD} build_packages
-			1>> ${BUILD_DIR}/${FULL_PROJECT_NAME}_make.log
-			2>> ${BUILD_DIR}/${FULL_PROJECT_NAME}_make.err
+			# 1>> ${BUILD_DIR}/${FULL_PROJECT_NAME}_make.log
+			# 2>> ${BUILD_DIR}/${FULL_PROJECT_NAME}_make.err
 		DEPENDS ${FULL_PROJECT_NAME}-make
 		WORKING_DIRECTORY ${BUILD_DIR}
 		COMMENT "Packaging ${FULL_PROJECT_NAME}"
-- 
2.8.1

