--- a/win32/CMakeLists.txt	2016-05-02 15:38:54.741181000 +0200
+++ b/win32/CMakeLists.txt	2016-05-02 15:35:19.684540700 +0200
@@ -43,9 +43,9 @@

 find_program( NSIS makensis PATH "c:\\Program Files (x86)\\NSIS" )

-if( NOT EXISTS "${NSIS}" )
-    message( FATAL_ERROR "NSIS is needed! Please install." )
-endif()
+# if( NOT EXISTS "${NSIS}" )
+#     message( FATAL_ERROR "NSIS is needed! Please install." )
+# endif()

 unigw_ParseVersion( WPKG ${UNIGW_VERSION} )

@@ -64,15 +64,15 @@
         ${CMAKE_CURRENT_SOURCE_DIR}/setup.nsi
 )

-add_custom_target( nsis_package ALL
-        ${NSIS_COMMMAND_LINE}
-        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-        DEPENDS build-all
-    )
-
-add_custom_target( nsis_package_standalone
-        ${NSIS_COMMMAND_LINE}
-        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-    )
+# add_custom_target( nsis_package ALL
+#         ${NSIS_COMMMAND_LINE}
+#         WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+#         DEPENDS build-all
+#     )
+#
+# add_custom_target( nsis_package_standalone
+#         ${NSIS_COMMMAND_LINE}
+#         WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+#     )

 # vim: ts=4 sw=4 et
