--- a/wpkg/package/CMakeLists.txt	2015-11-16 22:24:57.000000000 +0100
+++ b/wpkg/package/CMakeLists.txt	2016-05-02 16:50:12.502391800 +0200
@@ -41,18 +41,18 @@
 #    return()
 #endif()
 
-if( MO_MINGW32 )
-    get_filename_component( MINGW32_PATH ${CMAKE_MAKE_PROGRAM} DIRECTORY )
-    file( GLOB MINGW32_DLL_LIST "${MINGW32_PATH}/*.dll" )
-    install(
-        FILES
-            ${MINGW32_DLL_LIST}
-        DESTINATION
-            bin
-        COMPONENT
-            mingw32
-       )
-endif()
+# if( MO_MINGW32 )
+#     get_filename_component( MINGW32_PATH ${CMAKE_MAKE_PROGRAM} DIRECTORY )
+#     file( GLOB MINGW32_DLL_LIST "${MINGW32_PATH}/*.dll" )
+#     install(
+#         FILES
+#             ${MINGW32_DLL_LIST}
+#         DESTINATION
+#             bin
+#         COMPONENT
+#             mingw32
+#        )
+# endif()
 
 set( LOCAL_WPKG_ROOT "${CMAKE_BINARY_DIR}/WPKG" )
 if( UNIX )
@@ -73,14 +73,14 @@
     set( COMPONENT_LIST runtime development documentation headers )
     set( INSTALL_PACKAGES wpkg wpkg-dev wpkg-doc wpkg-hdrs-dev )
     set( PRIORITY required )
-    if( MO_WINDOWS )
-        set( PLATFORM_DEPENDS "Depends/runtime: msvc-runtime (= 12.0.0)" )
-    elseif( MO_MINGW32 )
-        set( SUBPACKAGES "${SUBPACKAGES}, mingw32" )
-        set( PLATFORM_DEPENDS "Depends/runtime: wpkgmingw32 (>= \${wpkg:version})" )
-        list( APPEND INSTALL_PACKAGES wpkgmingw32 )
-        list( APPEND COMPONENT_LIST mingw32 )
-    endif()
+#    if( MO_WINDOWS )
+#        set( PLATFORM_DEPENDS "Depends/runtime: msvc-runtime (= 12.0.0)" )
+#    elseif( MO_MINGW32 )
+#        set( SUBPACKAGES "${SUBPACKAGES}, mingw32" )
+#        set( PLATFORM_DEPENDS "Depends/runtime: wpkgmingw32 (>= \${wpkg:version})" )
+#        list( APPEND INSTALL_PACKAGES wpkgmingw32 )
+#        list( APPEND COMPONENT_LIST mingw32 )
+#    endif()
 endif()
 
 configure_file( admindb_init.txt "${LOCAL_WPKG_ROOT}/admindb_init.txt" @ONLY )
@@ -133,15 +133,15 @@
         WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
     )
 
-if( MO_WINDOWS )
-    set( MSVC_REDIST 
-        COMMAND ${WPKG}
-            --verbose
-            --compressor gzip
-            --output-dir "${WPKG_ROOT}/packages"
-            --build ${CMAKE_CURRENT_SOURCE_DIR}/vcredist
-    )
-endif()
+# if( MO_WINDOWS )
+#     set( MSVC_REDIST 
+#         COMMAND ${WPKG}
+#             --verbose
+#             --compressor gzip
+#             --output-dir "${WPKG_ROOT}/packages"
+#             --build ${CMAKE_CURRENT_SOURCE_DIR}/vcredist
+#     )
+# endif()
     
 add_custom_target( build_packages
         ${MSVC_REDIST}
@@ -167,6 +167,7 @@
             --root    "${WPKG_ROOT}"
             --running-copy
             --force-reinstall
+            --force-file-info
             --install ${INSTALL_PACKAGES}
         WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
         DEPENDS
