--- a/wpkg/libdebpackages/memfile.cpp	2015-11-16 22:24:57.000000000 +0100
+++ b/wpkg/libdebpackages/memfile.cpp	2017-05-25 13:35:51.867762461 +0200
@@ -299,9 +299,9 @@
     // I think we should have a command line flag so you can impose a limit
     // although there should be no reason other than package optimization
     // (i.e. make sure you don't include the "wrong" thing in your packages)
-    if(total > 1024 * 1024 * 1024)
+    if(total < 0)
     {
-        throw memfile_exception_parameter("memory file size too large (over 1Gb?!)");
+        throw memfile_exception_parameter("memory file size too large (over 2Gb?!)");
     }
 
     // allocate blocks to satisfy the total size
