--- a/wpkg/libdebpackages/wpkg_filename.cpp	2015-11-16 22:24:57.000000000 +0100
+++ b/wpkg/libdebpackages/wpkg_filename.cpp	2018-08-22 11:58:51.015391900 +0200
@@ -2369,8 +2369,8 @@
  */
 bool uri_filename::is_valid_windows_part(const std::string& path_part)
 {
-    const char *s(path_part.c_str());
-    const char *start(s);
+    const unsigned char *s(reinterpret_cast<const unsigned char *>(path_part.c_str()));
+    const unsigned char *start(s);
 
     if(*s == ' ')
     {
