diff --git a/wpkg/libdebpackages/libdebpackages/wpkg_output.h b/wpkg/libdebpackages/libdebpackages/wpkg_output.h
index ac3faa5..1e04e50 100644
--- a/wpkg/libdebpackages/libdebpackages/wpkg_output.h
+++ b/wpkg/libdebpackages/libdebpackages/wpkg_output.h
@@ -236,7 +236,7 @@ public:
     log& arg(const unsigned int v);
     log& arg(const signed long v);
     log& arg(const unsigned long v);
-#if defined(MO_WINDOWS) && defined(_WIN64) && !defined(MO_MINGW32)
+#if defined(MO_WINDOWS) && defined(_WIN64)
     log& arg(const size_t v);
 #endif
     log& arg(const float v);
--- a/wpkg/libdebpackages/wpkg_output.cpp	2015-03-06 15:17:47.000000000 +0100
+++ b/wpkg/libdebpackages/wpkg_output.cpp	2015-03-06 15:22:22.000000000 +0100
@@ -1599,13 +1599,13 @@
     ss << v;
     f_args.push_back(ss.str());
     return *this;
 }


-#if defined(MO_WINDOWS) && defined(_WIN64) && !defined(MO_MINGW32)
+#if defined(MO_WINDOWS) && defined(_WIN64)
 /** \brief Replace an argument with the size_t value.
  *
  * This function replaces an argument in the format string with the
  * numeric value of the specified parameter.
  *
  * \param[in] v  The value to replace the corresponding argument in the format string.
--- a/wpkg/libdebpackages/CMakeLists.txt	2015-03-06 15:06:03.000000000 +0100
+++ b/wpkg/libdebpackages/CMakeLists.txt	2015-03-06 14:59:49.000000000 +0100
@@ -43,13 +43,13 @@
 # WARNING: C/CXX flags are defined once for all the CMakeLists.txt projects
 if( MSVC )
     set( DEBPACKAGES_FOLDER libs/debpackages )
     # disable warnings about dll linkage concerns but we do not have any inlines (warning 4251)
     set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4251" )
 endif()
-if( MSVC OR MINGW32 OR MSYS )
+if( MSVC OR MINGW32 OR MINGW OR MSYS )
     set ( EXTRA_LIBRARIES wsock32 ws2_32 mpr ole32 uuid )
 endif()

 file( GLOB LIBDEBPACKAGES_INCLUDE_FILES "${CMAKE_CURRENT_SOURCE_DIR}/libdebpackages/*.h" )

 project( debpackages_static )
