--- a/wpkg/CMakeLists.txt 2018-09-27 15:51:42.482945274 +0200
+++ b/wpkg/CMakeLists.txt 2018-09-27 15:58:20.800056903 +0200
@@ -92,7 +92,7 @@
     execute_process( COMMAND uname -m OUTPUT_VARIABLE ARCH OUTPUT_STRIP_TRAILING_WHITESPACE )
     if( (${ARCH} MATCHES i386) OR (${ARCH} MATCHES i686) )
         set( WPKG_PROCESSOR i386 )
-    else( ${ARCH} MATCHES x86_64 )
+    elseif( ${ARCH} MATCHES x86_64 )
         set( WPKG_PROCESSOR amd64 )
     else()
         message( FATAL_ERROR "Unknown architecture '${ARCH}'!" )
