--- a/wpkg/CMakeLists.txt	2019-04-16 17:49:30.000000000 +0200
+++ b/wpkg/CMakeLists.txt	2019-05-02 09:10:19.000000000 +0200
@@ -130,11 +130,11 @@
                 set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnoexcept")
             endif()
         endif()
-        set( OPT "-O3" )
-        if( MO_DARWIN )
-            message( WARNING "Disabling optimization for DARWIN, since the compiler generates code that crashes pkg_explorer! TODO: this needs to be addressed!" )
-            set( OPT "-O0" )
-        endif()
+        set( OPT "-O3 -mtune=native" )
+        # if( MO_DARWIN )
+        #     message( WARNING "Disabling optimization for DARWIN, since the compiler generates code that crashes pkg_explorer! TODO: this needs to be addressed!" )
+        #     set( OPT "-O0" )
+        # endif()
         set( CMAKE_CXX_FLAGS_DEBUG   "${CMAKE_CXX_FLAGS_DEBUG}   -O0"    )
         set( CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${OPT}" )
     endif()
