--- a/wpkg/libdebpackages/wpkgar_build.cpp	Tue Apr 28 12:17:31 2015
+++ b/wpkg/libdebpackages/wpkgar_build.cpp	Tue Apr 28 12:20:45 2015
@@ -2866,6 +2866,13 @@
                     cmd += " --create-index index.tar.gz";
                     cmd += " --force-file-info";
                     cmd += " --run-unit-tests";
+                    cmd += " --make-tool ";
+                    cmd += wpkg_util::make_safe_console_string(f_make_tool);
+                    if(!f_cmake_generator.empty())
+                    {
+                        cmd += " --cmake-generator ";
+                        cmd += wpkg_util::make_safe_console_string(f_cmake_generator);
+                    }
 
                     // keep the same debug flags for sub-calls
                     cmd += " --debug ";
