diff --git a/wpkg/libdebpackages/wpkgar_build.cpp b/wpkg/libdebpackages/wpkgar_build.cpp
index e555618..aaf8994 100644
--- a/wpkg/libdebpackages/wpkgar_build.cpp
+++ b/wpkg/libdebpackages/wpkgar_build.cpp
@@ -2810,6 +2810,7 @@ void wpkgar_build::build_repository()
                 if(ready)
                 {
                     // this source package is ready to get built
+                    bool addToRepository = false;
                     std::string cmd(f_program_fullname);
                     cmd += " ";
                     cmd += " --root ";
@@ -2820,10 +2821,24 @@ void wpkgar_build::build_repository()
                     cmd += f_manager->get_database_path().full_path();
                     cmd += " --build ";
                     cmd += sources[i]->f_filename.full_path();
-                    cmd += " --repository ";
-                    cmd += f_build_directory.full_path();
                     cmd += " --output-repository-dir ";
+                    if(!f_output_repository_dir.empty())
+                    {
+                        cmd += f_output_repository_dir.full_path();
+
+                        wpkg_filename::uri_filename::file_stat s;
+                        addToRepository = f_output_repository_dir.os_stat(s) == 0;
+                    }
+                    else
+                    {
+                        cmd += f_build_directory.full_path();
+                    }
+                    cmd += " --repository ";
                     cmd += f_build_directory.full_path();
+                    if(addToRepository)
+                    {
+                        cmd += " " + f_output_repository_dir.full_path();
+                    }
                     cmd += " --create-index index.tar.gz";
                     cmd += " --force-file-info";
                     cmd += " --run-unit-tests";
