--- a/wpkg/libdebpackages/wpkgar_install.cpp	2020-02-03 11:46:35.310490375 +0100
+++ b/wpkg/libdebpackages/wpkgar_install.cpp	2020-02-03 11:47:29.120295931 +0100
@@ -2473,6 +2473,24 @@
 {
     const auto filename( item.get_filename() );
 
+    bool skip( false );
+    f_manager->load_package("core");
+    const std::string target_architecture = f_manager->get_field("core", "Architecture");
+
+    for(std::vector<std::string>::size_type k(0); k < dependency.f_architectures.size(); ++k)
+    {
+        if(!wpkg_dependencies::dependencies::match_architectures(target_architecture, dependency.f_architectures[k]))
+        {
+            skip = true;
+            break;
+        }
+    }
+
+    if( skip )
+    {
+        return false;
+    }
+
     // if an explicit package has a dependency satisfied by another
     // explicit package then we mark all implicit packages of the same
     // name as invalid because they for sure won't get used
