--- a/wpkg/libdebpackages/wpkg_filename.cpp	Tue Apr 28 12:17:31 2015
+++ b/wpkg/libdebpackages/wpkg_filename.cpp	Tue Apr 28 12:20:45 2015
@@ -3543,7 +3543,8 @@
                 r = unlink(unlink_filename.os_filename().get_os_string().c_str());
                 f_errno = errno;
                 // Under MS-Windows we may get an EACCESS error instead of EISDIR
-                if(r != 0 && (f_errno == EISDIR || f_errno == EACCES))
+                // Under darwin, we get an EPERM error when it's a directory
+                if(r != 0 && (f_errno == EISDIR || f_errno == EACCES || f_errno == EPERM))
                 {
                     // we need an rmdir() for a directory
                     r = rmdir(unlink_filename.os_filename().get_os_string().c_str());
