--- a/wpkg/libdebpackages/wpkg_filename.cpp	Fri Mar 13 13:33:06 2015
+++ b/wpkg/libdebpackages/wpkg_filename.cpp	Fri Mar 13 13:33:16 2015
@@ -3932,7 +3932,7 @@
     wpkg_filename::uri_filename::os_filename_t cwd("...undefined folder...");
     wpkg_filename::uri_filename::os_char_t buf[32 * 1024]; // limit is 32Kb under MS-Windows
 #if defined(MO_MINGW32) || defined(MO_MINGW64)
-    if(GetCurrentDirectoryW(sizeof(buf) / sizeof(buf[0]), buf) == 0)
+    if(GetCurrentDirectoryW(sizeof(buf) / sizeof(buf[0]), buf) != 0)
 #else
     if(::getcwd(buf, sizeof(buf) / sizeof(buf[0])) != NULL)
 #endif
