--- a/wpkg/libdebpackages/wpkgar_build.cpp	2020-02-06 07:04:33.685839698 +0100
+++ b/wpkg/libdebpackages/wpkgar_build.cpp	2020-02-06 08:12:58.353956922 +0100
@@ -1005,6 +1014,19 @@
         f_package_name = f_output_dir.append_child(f_package_name.path_only());
     }
     debian_ar.write_file(f_package_name, true);
+
+    // generate MD5 sum by the way, it can be used by the create_index command
+    // in order to improve the index generation speed.
+    md5::raw_md5sum raw;
+    debian_ar.raw_md5sum(raw);
+
+    memfile::memory_file md5sum_file;
+    md5sum_file.create(memfile::memory_file::file_format_other);
+    md5sum_file.printf("%s", md5::md5sum::sum(raw).c_str());
+
+    wpkg_filename::uri_filename md5sum_filename(f_package_name);
+    md5sum_filename.set_filename(md5sum_filename.original_filename() + ".md5sum");
+    md5sum_file.write_file(md5sum_filename);
 }
 
 