--- a/wpkg/libdebpackages/wpkgar_build.cpp	Fri Mar 13 14:45:08 2015
+++ b/wpkg/libdebpackages/wpkgar_build.cpp	Fri Mar 13 14:46:32 2015
@@ -2004,7 +2004,11 @@
     prepare_cmd(cmd, build_tmpdir);
     if((wpkg_output::get_output_debug_flags() & wpkg_output::debug_flags::debug_progress) != 0)
     {
+#if defined(MO_WINDOWS)
+        cmd += "set VERBOSE=1 && ";
+#else
         cmd += "VERBOSE=1 ";
+#endif
     }
     cmd += wpkg_util::make_safe_console_string(f_make_tool);
     cmd += " package_source";
@@ -2299,7 +2303,11 @@
     prepare_cmd(make_all_cmd, build_tmpdir);
     if((wpkg_output::get_output_debug_flags() & wpkg_output::debug_flags::debug_progress) != 0)
     {
+#if defined(MO_WINDOWS)
+        make_all_cmd += "set VERBOSE=1 && ";
+#else
         make_all_cmd += "VERBOSE=1 ";
+#endif
     }
     make_all_cmd += wpkg_util::make_safe_console_string(f_make_tool);

@@ -2349,7 +2357,11 @@
     prepare_cmd(run_tests_cmd, build_tmpdir);
     if((wpkg_output::get_output_debug_flags() & wpkg_output::debug_flags::debug_progress) != 0)
     {
+#if defined(MO_WINDOWS)
+        run_tests_cmd += "set VERBOSE=1 && ";
+#else
         run_tests_cmd += "VERBOSE=1 ";
+#endif
     }
     run_tests_cmd += wpkg_util::make_safe_console_string(f_make_tool);
     run_tests_cmd += " run_unit_tests";
