--- a/wpkg/libdebpackages/wpkgar_build.cpp	Fri Mar 13 15:21:55 2015
+++ b/wpkg/libdebpackages/wpkgar_build.cpp	Fri Mar 13 15:44:02 2015
@@ -1851,7 +1851,11 @@
  */
 void wpkgar_build::prepare_cmd(std::string& cmd, const wpkg_filename::uri_filename& dir)
 {
+#if defined(MO_WINDOWS)
+    cmd = "cd /d ";
+#else
     cmd = "cd ";
+#endif
     cmd += wpkg_util::make_safe_console_string(dir.full_path());
 #if defined(MO_WINDOWS)
     cmd += " && set PATH=";

@@ -2445,7 +2445,12 @@
         created_packages[sub_name] = true;

         // run cmake
-        std::string cmake_cmd("cd " + build_tmpdir.full_path() + " && cmake");
+#if defined(MO_WINDOWS)
+        std::string cmake_cmd("cd /d ");
+#else
+        std::string cmake_cmd("cd ");
+#endif
+        cmake_cmd += build_tmpdir.full_path() + " && cmake";
         wpkg_filename::uri_filename install_dir(install_tmpdir.append_child(sub_name).append_safe_child(f_install_prefix));
         cmake_cmd += " -DCMAKE_INSTALL_PREFIX=" + install_dir.full_path();
         cmake_cmd += " -DCMAKE_INSTALL_COMPONENT=" + sub_name;
