diff --git a/wpkg/libdebpackages/libdebpackages/wpkg_output.h b/wpkg/libdebpackages/libdebpackages/wpkg_output.h
index ac3faa5..be98568 100644
--- a/wpkg/libdebpackages/libdebpackages/wpkg_output.h
+++ b/wpkg/libdebpackages/libdebpackages/wpkg_output.h
@@ -236,7 +236,7 @@ public:
     log& arg(const unsigned int v);
     log& arg(const signed long v);
     log& arg(const unsigned long v);
-#if defined(MO_WINDOWS) && defined(_WIN64) && !defined(MO_MINGW32)
+#if defined(MO_WINDOWS) && defined(_WIN64) && (!defined(MO_MINGW32) || defined(__MINGW64__))
     log& arg(const size_t v);
 #endif
     log& arg(const float v);
diff --git a/wpkg/libdebpackages/wpkg_output.cpp b/wpkg/libdebpackages/wpkg_output.cpp
index f345089..6843945 100644
--- a/wpkg/libdebpackages/wpkg_output.cpp
+++ b/wpkg/libdebpackages/wpkg_output.cpp
@@ -1598,7 +1598,7 @@ log& log::arg(const unsigned long v)
 }
 
 
-#if defined(MO_WINDOWS) && defined(_WIN64) && !defined(MO_MINGW32)
+#if defined(MO_WINDOWS) && defined(_WIN64) && (!defined(MO_MINGW32) || defined(__MINGW64__))
 /** \brief Replace an argument with the size_t value.
  *
  * This function replaces an argument in the format string with the
